v22.1/cluster-setup-troubleshooting.md

```diff
@@ -561,6 +561,25 @@ The [DB Console][db_console] provides several ways to check for node liveness is
 For more information about how node liveness works, see [Replication Layer](architecture/replication-layer.html#epoch-based-leases-table-data).
 {{site.data.alerts.end}}
 
+#### Impact of node failure is greater than 10 seconds
+
+When a node fails its liveness check, [each of its leases is transferred to a healthy node](architecture/replication-layer.html#how-leases-are-transferred-from-a-dead-node
+). In theory, this process should take no more than 9 seconds for liveness expiration plus the cost of 2 network roundtrips: 1 for Raft leader election, and 1 for lease acquisition.
+
+In production, depending on the version of CockroachDB you are running, lease transfer upon node failure can be "non-cooperative" and can take longer than expected. This is observed in the following scenarios:
+
+- **The leaseholder node for the liveness range fails.** The liveness range is a system range that [stores the liveness record](architecture/replication-layer.html#epoch-based-leases-table-data) for each node on the cluster. If a node fails and is also the leaseholder for the liveness range, operations cannot proceed until the liveness range is transferred to a new leaseholder and the liveness record is made available to other nodes. This can cause momentary cluster unavailability.
+
+- **The leaseholder node for `system.users` fails.** The cluster accesses the `system.users` table when authenticating a user. Prior to v21.2.0, a customer attempting to log in to the cluster would encounter a delay while waiting for the `system.users` table to find a new leaseholder. In v21.2.0 and later, there is no login delay because the authentication data is cached.
+
+- **A node comes in and out of liveness, due to network failures or overload.** Prior to version TK, a "flapping" node could repeatedly lose and acquire leases in between its intermittent heartbeats, causing wide-ranging unavailability. In versions TK and later, a node must successfully heartbeat for 30 seconds after failing a heartbeat before it is eligible to acquire leases.
+
+- **Network issues cause connection issues between nodes or DNS.** Prior to v20.2.18 and v21.1.12, this could cause 10 to 60 seconds of unavailability as the system stalled on network throughput, preventing a speedy movement of leases and recovery. In subsequent versions, CockroachDB avoids contacting unresponsive nodes or DNS during certain performance-critical operations.
```

Comment:
TFTR Erik! I tried to cover this in a new, separate bullet. But now I wonder if the "Network issues cause connection issues between nodes or DNS." bullet needs to be rephrased to sound more distinct. What do you think?

About the network partition, do you mean an asymmetric network partition, or is asynchronous another type? If the former, we have a pretty light [section](https://www.cockroachlabs.com/docs/v22.1/cluster-setup-troubleshooting.html#network-partition) on the same page. It is overdue for an update. I have a high-priority [ticket](https://cockroachlabs.atlassian.net/browse/DOC-2850) to expand our network troubleshooting docs, and @mwang1026 has suggested that this could focus on asymmetric network partitions, and how to proactively look for them. So I think I should save your latter example for this issue. If you have further details (on what happens in these network partition scenarios, and how one might address), please feel free to comment in that ticket.

Reply:
I added these across the docs versions - let me know how they look. 

Regarding circuit breakers, is that the same feature documented here: https://www.cockroachlabs.com/docs/v22.1/architecture/replication-layer.html#per-replica-circuit-breakers

If so, do you mean to emphasize that the circuit breakers introduced in v22.1 can usually throw errors (good), or that they sometimes try to connect and don't return an error (bad)?